### ISSUE TITLE
Found simulator backdoor for passing Term2 particle filter project

### ISSUE THREAD

**Issue by AeroGeekDean** [udacity/sdc-issue-reports#993]
I accidentally forgot to set is_initialized = true in ParticleFilter::init(). This lead to my filter to run in init mode on every frame of the sim run, and my filter was able to magically pass project criteria test even with num_particles = 1 !!!

Thus thru deduction, my particle filter process is receiving initialization messages from the sim on every frame. These messages are:
- 'sense_x'
- 'sense_y'
- 'sense_theta'

Additionally, these messages are most likely being filled with the latest state info (ground truth) of the simulator vehicle. Thus likely how my code (with bug) was able to pass the Kidnapped Vehicle project criteria.

**Comment by baumanab** [comment]
Nice work finding this potential exploit.  Potential exploits such as this are the reason for the final rubric point requiring actual use of a particle filter for localization.  Please let us know if you find any more potential exploits or more details on this one.

**Comment by AeroGeekDean** [comment]
Actually, someone could have a particle filter that ***looks*** correct, but actually be non-functional because of maybe bugs within their algorithm. Now that I [posted it on slack](https://carnd.slack.com/archives/C4TSP7BFU/p1501996150006054) (a few weeks ago), people who couldn't get their filter to work might be tempted to use this exploit... Since even my project reviewer couldn't figure it out.

I stumbled across this 'exploit' because I was curious about the effect of minimizing the number of particles, trying to see how the filter will break down as I reduce `num_particles`. To my surprise, my filter was able to successfully pass project criteria even with just ONE particle. Repeatedly. This seemed insanely improbable, until I re-walked through my code and noticed that I forgot to set the initialization flag. Hence this is how I deduced about the 'exploit'.

Not sure what other details would help Udacity close off this exploit?
My code base is here: https://github.com/AeroGeekDean/CarND-Kidnapped-Vehicle-Project

Perhaps a quick fix might be to modify the simulator (for this project) such it does NOT send initialization data after the first few frames? These data would be the `sense_x`, `sense_y`, `sense_theta` messages. Maybe allow these to be send only during the first X-number of frames (in case IPC messages got dropped), but then disable sending them after that. Just a thought...
